src/main/scala/cromwell/engine/backend/local/SharedFileSystem.scala

```diff
@@ -56,14 +60,15 @@ object SharedFileSystem {
    * indirectly through one of its input expressions
    */
   private def localizePathViaSymbolicLink(call: Option[Call], originalPath: Path, executionPath: Path): Try[Unit] = {
-    call.flatMap(_.docker) match {
-      case Some(_) => Failure(new UnsupportedOperationException("Cannot localize with symbolic links with Docker"))
-      case _ => Try(Files.createSymbolicLink(executionPath, originalPath.toAbsolutePath))
-    }
+    if (call.flatMap(_.docker).isDefined)
+      Failure(new UnsupportedOperationException("Cannot localize with symbolic links with Docker"))
+    else if (originalPath.toFile.isDirectory)
+      Failure(new UnsupportedOperationException("Cannot localize directory with symbolic links"))
+    else Try(Files.createSymbolicLink(executionPath, originalPath.toAbsolutePath))
   }
 }
 
-trait SharedFileSystem {
+trait SharedFileSystem extends Backend {
```

Comment:
@Horneth That's what concerned me. I don't think it's a good idea to be slapping on traits out of convenience, it should imply either "is-a" or "has-this-functionality" (i.e. mixin). The self-type as @mcovarr is something in between an is-a and a has-a. I don't really think any of the above really apply - I think your second paragraph is heading in the correct direction (if it's not already correct), identifying what the shared component is here and split that out.